GameEngine/Characters/Character.cs

```diff
@@ -86,6 +91,34 @@ private int GetAttackDamage(int maxAttack)
             return damage;
         }
 
-        //ZABTODO: Make a heal fuction and add a healing power method
+        public bool IsDead()
+        {
+            if(Hp <= 0)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public void Heal(Character healingCharacter, GameSourceData gameData)
+        {
+            var healAmount = GetHealAmount(MaxHeal);
+            var charactersInLocation = GetCharacters(gameData);
+            Hp = Hp + healAmount;
```

Comment:
Technically you probably want to do if ((Hp + healAmount) > FullHp) { healAmount = FullHp - Hp } 
Then do the heal and say the message.
That way you never actually have the HP go over the limit, instead of correcting it after.